cli/cmd/plugin/unmanaged-cluster/cluster/minikube.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/config"
+)
+
+var minikubeBinName = "minikube"
+
+type MinikubeClusterManager struct{}
+
+// Create
+func (mkcm MinikubeClusterManager) Create(c *config.UnmanagedClusterConfig) (*KubernetesCluster, error) {
+	if !mkcm.isMinikubeRunning() {
+		// TODO: Is minikube it the current context? If not, attempt to switch to it
+
+		args := []string{
+			"start",
+		}
+
+		cmd := exec.Command(minikubeBinName, args...)
+
+		err := cmd.Run()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// readkubeconfig in bytes from configured kubeconfig.
+	// During `Prepare` this should be set to the users home kubeconfig since that's what minikube automatically uses
+	kcBytes, err := os.ReadFile(c.KubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	kc := &KubernetesCluster{
+		Name:       c.ClusterName,
+		Kubeconfig: kcBytes,
+	}
+
+	return kc, nil
+}
+
+// isMinikubeRunning returns true if `minikube status` is successful
+// Running `minikube status` returns an error if the node is stopped or no profile is found
+// In those cases, we can attempt to recreate
+func (mkcm MinikubeClusterManager) isMinikubeRunning() bool {
+	args := []string{
+		"status",
+	}
+
+	cmd := exec.Command(minikubeBinName, args...)
+
+	_, err := cmd.Output()
+
+	return err == nil
+}
+
+// Get returns cluster information or returns an error
+func (mkcm MinikubeClusterManager) Get(clusterName string) (*KubernetesCluster, error) {
+	return nil, nil
```

Comment:
Guessing this is TODO, small note, let's leverage looking up profile details.

Reply:
What is the point of this API? We don't use it in the `kind` provider and we don't have a top level command (like `unmanaged-cluster get {cluster-name}` that uses this.

https://github.com/vmware-tanzu/community-edition/blob/30e52bfc597e19b337e9349bd80e05b7cc297f4e/cli/cmd/plugin/unmanaged-cluster/cluster/kind.go#L271-L274

Maybe this is something we punt on until we bring in a `get` top level command?

Reply:
Fine to punt for now, add a `TODO`?